dissect/target/plugins/os/unix/bsd/darwin/ios/_os.py

```diff
@@ -0,0 +1,95 @@
+from dissect.target.plugins.os.unix.bsd.darwin._os import (
+    DarwinPlugin,
+    detect_macho_arch,
+)
+from dissect.target.target import Target
+
+
+class IOSPlugin(DarwinPlugin):
+    """Apple iOS plugin.
+
+    Resources:
+        - https://developer.apple.com/library/archive/documentation/FileManagement/Conceptual/FileSystemProgrammingGuide/FileSystemOverview/FileSystemOverview.html
+        - https://corp.digitalcorpora.org/corpora/mobile/iOS17/
+    """  # noqa: E501
+
+    SYSTEM = "/private/var/preferences/SystemConfiguration/preferences.plist"
+    GLOBAL = "/private/var/mobile/Library/Preferences/.GlobalPreferences.plist"
+    VERSION = "/System/Library/CoreServices/SystemVersion.plist"
+
+    # /private/etc/master.passwd is a copy of /private/etc/passwd
+    PASSWD_FILES = ["/private/etc/passwd"]
+
+    def __init__(self, target: Target):
+        super().__init__(target)
+
+        self._config = Config(
+            target.fs.path(self.SYSTEM),
+            target.fs.path(self.GLOBAL),
+            target.fs.path(self.VERSION),
+        )
+
+    @classmethod
+    def detect(cls, target: Target) -> Filesystem | None:
+        for fs in target.filesystems:
+            if fs.exists("/private/var/preferences") and fs.exists("/private/var/mobile"):
+                return fs
+
+    @classmethod
+    def create(cls, target: Target, sysvol: VirtualFilesystem) -> None:
+        target.fs.mount("/", sysvol)
+        return cls(target)
+
+    @export(property=True)
+    def hostname(self) -> str | None:
+        try:
+            # ComputerName can contain invalid utf characters, so we use HostName instead.
+            return self._config.SYSTEM["System"]["System"]["HostName"]
+        except KeyError:
+            pass
+
+    @export(property=True)
+    def ips(self) -> list:
+        return []
+
+    @export(property=True)
+    def version(self) -> str:
+        return f'{self._config.VERSION["ProductName"]} {self._config.VERSION["ProductVersion"]} ({self._config.VERSION["ProductBuildVersion"]})'  # noqa: E501
+
+    @export(record=IOSUserRecord)
+    def users(self) -> Iterator[IOSUserRecord]:
+        for user in super().users():
+            yield IOSUserRecord(**user._asdict())
+
+    @export(property=True)
+    def os(self) -> str:
+        return OperatingSystem.IOS.value
+
+    @export(property=True)
+    def architecture(self) -> str | None:
+        if arch := detect_macho_arch(["/bin/df", "/bin/ps", "/sbin/fsck", "/sbin/mount"], fs=self.target.fs):
+            return f"{arch}-ios"
+
+
+@dataclass
+class Config:
+    SYSTEM: dict[str, Any]
+    GLOBAL: dict[str, Any]
+    VERSION: dict[str, Any]
+
+    def __post_init__(self):
+        for field in self.__dataclass_fields__.keys():
+            path = getattr(self, field)
+            if path.is_file():
+                setattr(self, field, plistlib.load(path.open("rb")))
+            else:
+                setattr(self, field, {})
```

Comment:
This didn't change the problem that `Config` will seem to have an `__init__` that looks like this:

```python
class Config
    def __init__(self, SYSTEM: dict[str, Any], GLOBAL: dict[str, Any], VERSION: dict[str, Any]): ...
```

And then you initialize it with 3 `TargetPath` objects instead. You fix it at runtime in `__post_init__`, but I want to prevent we merge something like this now and have to completely change it later when we will apply stronger type hinting to this project.

Reply:
That'll also be a huge pain with a type checker, since it will always complain when you try to access an attribute on `dict` that doesn't exist on `Path`, you'd need to assert it was a `dict` first.

Maybe better to use a `classmethod` initializer like `Config.load(path1, path2, path2)`.